libcontainer/factory_linux_test.go

```diff
@@ -47,6 +47,10 @@ func TestFactoryNew(t *testing.T) {
 	if factory.Type() != "libcontainer" {
 		t.Fatalf("unexpected factory type: %q, expected %q", factory.Type(), "libcontainer")
 	}
+
+	if lfactory.CriuPath != "criu" {
+		t.Fatalf("unexpected factory CriuPath: %q, expected %q", lfactory.CriuPath, "criu")
```

Comment:
I think `defer syscall.Unmount(root, syscall.MNT_DETACH)` at last line of this function is wrong,by the way add `lfactory.CriuPath` check, like `factory.Type() != "libcontainer"`